docs/content/docs/deployment/elastic_scaling.md

```diff
@@ -0,0 +1,107 @@
+
+The Reactive Mode allows Flink users to implement a powerful autoscaling mechanism, by having an external service monitor certain metrics, such as consumer lag, aggregate CPU utilization, throughput or latency. As soon as these metrics are above or below a certain threshold, additional TaskManagers can be added or removed from the Flink cluster. This could be implemented through changing the [replica factor](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#replicas) of a Kubernetes deployment, or an [autoscaling](https://docs.aws.amazon.com/autoscaling/ec2/userguide/AutoScalingGroup.html) group. This external service only needs to handle the resource allocation and deallocation. Flink will take care of keeping the job running with the resources available.
+ 
+### Getting started
+
+If you just want to try out Reactive Mode, follow these instructions. They assume that you are deploying Flink on one machine.
+
+```bash
+
+# these instructions assume you are in the root directory of a Flink distribution.
+
+# Put Job into lib/ directory
+cp ./examples/streaming/TopSpeedWindowing.jar lib/
+# Submit Job in Reactive Mode
+./bin/standalone-job.sh start -Drestart-strategy=fixeddelay -Drestart-strategy.fixed-delay.attempts=100000 -Dscheduler-mode=reactive -j org.apache.flink.streaming.examples.windowing.TopSpeedWindowing
+# Start first TaskManager
+./bin/taskmanager.sh start
+```
+
+Let's quickly examine the used submission command:
+- `./bin/standalone-job.sh start` deploys Flink in [Application Mode]({{< ref "docs/deployment/overview" >}}#application-mode)
+- `-Drestart-strategy=fixeddelay` and `-Drestart-strategy.fixed-delay.attempts=100000` configure the job to restart on failure. This is needed for supporting scale-down.
+- `-Dscheduler-mode=reactive` enables Reactive Mode.
+- the last argument is passing the Job's main class name.
+
+You have now started a Flink job in Reactive Mode. The [web interface](http://localhost:8081) shows that the job is running on one TaskManager. If you want to scale up the job, simply add another TaskManager to the cluster:
+```bash
+# Start additional TaskManager
+./bin/taskmanager.sh start
+```
+
+To scale down, remove a TaskManager instance.
+```bash
+# Remove a TaskManager
+./bin/taskmanager.sh stop
+```
+
+### Usage
+
+#### Configuration
+
+To enable Reactive Mode, you need to configure `scheduler-mode` to `reactive`.
+
+#### Parallelism
+
+The **parallelism of individual operators in a job will be determined by the scheduler**. It is not configurable. 
+
+The only way of influencing the parallelism is by setting a max parallelism for a operator (which will be respected by the scheduler). The maxParallelism is bounded by 2^15 (32768), which is the value that Reactive Mode uses if nothing else is configured. If there is no maxParallelism defined for an operator, `32768` will be used as a default.
```

Comment:
Without Reactive Mode the maxParallelism default to `128`, correct?

I am wondering if Reactive mode should change the maxParallelism setting implicitly. It feels more naturally to me, if the maxParallelism is not changed and configured as an orthogonal concern by the user regardless of the scheduler-mode.

Reply:
Correct, the default is 128.

I don't understand your suggestion. The current behavior is implicit in my opinion: If the user does nothing, they can scale up the job in reactive mode up to a parallelism of 32768.

Reply:
Yes, I am wondering if this implicitness is a bad thing. I would suggest not to change it and to let the user increase the max parallelism explicitly if she wants to. 

Thinking more about this: in the current way, wouldn't changing the scheduler-mode break savepoint compatibility right now? You switch a job from default to reactive and you can't recover from your Savepoint by default?! 

But even if that's not the case, I find it more natural not to implicitely change max parallelism when changing the scheduler mode. The max parallelism is respected by the schedule (as it already is), but not changed.

Reply:
Since this already implemented as I've documented it: How about merging the PR as-is, because it is in sync with the code, then we revisit this issue in a separate ticket?

This probably breaks savepoint compatibility, yes.
It also leads to poor performance, because we have so many key groups. You generally have compelling arguments to revisit this.

Reply:
https://issues.apache.org/jira/browse/FLINK-21844

Reply:
Thank you for filing a ticket!